panic: runtime error: assignment to entry in nil map

```
panic: runtime error: assignment to entry in nil map

goroutine 16 [running]:
runtime.panic(0x59d7e0, 0x6befb3)
        /usr/local/go/src/pkg/runtime/panic.c:279 +0xf5
main.(*MdWriter).writeMarkdownFromTable(0xc208a95220, 0x2b95842cfb18, 0xc208a95218, 0xc2087285a0, 0x0, 0x0)
        /home/vagrant/src/github.com/yuya-takeyama/dbyaml2md/md_writer.go:51 +0x10d
main.generateMarkdownFiles(0xc2088d9b50, 0x0, 0x0)
        /home/vagrant/src/github.com/yuya-takeyama/dbyaml2md/main.go:96 +0x37c
main.func·001(0xc20804c160)
        /home/vagrant/src/github.com/yuya-takeyama/dbyaml2md/main.go:75 +0x314
github.com/codegangsta/cli.(*App).Run(0xc20804a0c0, 0xc20800e000, 0x2, 0x2, 0x0, 0x0)
        /home/vagrant/src/github.com/codegangsta/cli/app.go:130 +0x98a
main.main()
        /home/vagrant/src/github.com/yuya-takeyama/dbyaml2md/main.go:78 +0x222

goroutine 19 [finalizer wait]:
runtime.park(0x4146e0, 0x6c2b08, 0x6c1649)
        /usr/local/go/src/pkg/runtime/proc.c:1369 +0x89
runtime.parkunlock(0x6c2b08, 0x6c1649)
        /usr/local/go/src/pkg/runtime/proc.c:1385 +0x3b
runfinq()
        /usr/local/go/src/pkg/runtime/mgc0.c:2644 +0xcf
runtime.goexit()
        /usr/local/go/src/pkg/runtime/proc.c:1445
```

--- a/md_writer.go
+++ b/md_writer.go
@@ -67,6 +67,10 @@ func (mdWriter *MdWriter) writeMarkdown(file io.Writer, table *model.Table) erro
 
 func (mdWriter *MdWriter) frontMatterWithTable(table *model.Table) FrontMatter {
 	frontMatter := mdWriter.frontMatter
+	if frontMatter == nil {
+		frontMatter = make(FrontMatter)
+	}
+
 	frontMatter["table"] = table
 
 	return frontMatter

--- a/md_writer_test.go
+++ b/md_writer_test.go
@@ -170,3 +170,38 @@ Users table
 
 	assert.Contains(t, buf.String(), expectedPrefix)
 }
+
+func TestIssue6IfFrontMatterIsNil(t *testing.T) {
+	var frontMatter FrontMatter
+	frontMatter = nil
+
+	table := &model.Table{
+		Name:    "users",
+		Columns: make([]*model.Column, 0),
+		Indexes: make([]*model.Index, 0),
+		Comment: "Users table",
+		DDL:     "CREATE TABLE users",
+	}
+
+	buf := new(bytes.Buffer)
+	mdWriter := &MdWriter{frontMatter}
+	err := mdWriter.writeMarkdown(buf, table)
+	if err != nil {
+		t.Fatalf("Failed to write generated markdown into buffer: %s", err)
+	}
+
+	expectedPrefix := `---
+table:
+  name: users
+  columns: []
+  indexes: []
+  comment: Users table
+  ddl: CREATE TABLE users
+---
+# users
+
+Users table
+`
+
+	assert.Contains(t, buf.String(), expectedPrefix)
+}

